### PR TITLE
WB-320: Add tests for 'clientNav' prop on Button, IconButton, and Link

### PR DESCRIPTION
**Pull request by kevinbarabash**
The tests actually picked up an issue with the addition of `clientNav` to `IconButton` in #239.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 26d80e520ab9b731feb36e0d10d52e5bb29753d3

https://deploy-preview-240--wonder-blocks.netlify.com

File: packages/wonder-blocks-button/components/button.test.js [hunk 1]
```diff
@@ -1,14 +1,58 @@
...
-    test("render a label", (done) => {
-        const wrapper = shallow(
-            <Button onClick={() => done()}>Hello World!</Button>,
+    beforeEach(() => {
+        unmountAll();
+    });
+
+    test("client-side navigation", () => {
+        const wrapper = mount(
+            <MemoryRouter>
```
Comment (kevinbarabash) [inline packages/wonder-blocks-button/components/button.test.js]: `MemoryRouter` handles all route changes in memory making it ideal for testing.

File: packages/wonder-blocks-button/components/button.test.js [hunk 2]
```diff
@@ -1,14 +1,58 @@
...
+                    </Switch>
+                </div>
+            </MemoryRouter>,
         );
-        wrapper.simulate("click");
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper.find(`[data-test-id="button"]`).first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
```
Comment (kevinbarabash) [inline packages/wonder-blocks-button/components/button.test.js]: The `{button: 0}` is necessary b/c `react-router-dom`'s `Link` component checks for it.

File: packages/wonder-blocks-icon-button/components/icon-button.js [hunk 3]
```diff
@@ -150,6 +150,7 @@ export default class IconButton extends React.Component<SharedProps> {
                             {...sharedProps}
                             {...state}
                             {...handlers}
+                            clientNav={clientNav}
```
Comment (kevinbarabash) [inline packages/wonder-blocks-icon-button/components/icon-button.js]: whoops

File: packages/wonder-blocks-button/components/button.test.js [hunk 4]
```diff
@@ -1,14 +1,58 @@
 // @flow
 import React from "react";
-import {shallow} from "enzyme";
+import {MemoryRouter, Route, Switch} from "react-router-dom";
 
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
```
Comment (somewhatabstract) [inline packages/wonder-blocks-button/components/button.test.js]: 😻

File: packages/wonder-blocks-button/components/button.test.js (hunk 2 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-button/components/button.test.js]: I'd love to see `// Arrange`, `// Act`, `// Assert` structure in these tests to make it clear what the test is doing.

File: packages/wonder-blocks-button/components/button.test.js [hunk 5]
```diff
@@ -1,14 +1,58 @@
...
+                    </Switch>
+                </div>
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper.find(`[data-test-id="button"]`).first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(false);
```
Comment (somewhatabstract) [inline packages/wonder-blocks-button/components/button.test.js]: I'd love to see `// Arrange`, `// Act`, `// Assert` structure in these tests to make it clear what the test is doing.

File: packages/wonder-blocks-icon-button/components/icon-button.js (hunk 3 above)
Comment (somewhatabstract) [reply packages/wonder-blocks-icon-button/components/icon-button.js]: Nice catch! Testing FTW!

File: packages/wonder-blocks-icon-button/components/icon-button.test.js [hunk 6]
```diff
@@ -1,12 +1,18 @@
 // @flow
 import React from "react";
-import {mount, shallow} from "enzyme";
+import {shallow} from "enzyme";
+import {MemoryRouter, Route, Switch} from "react-router-dom";
 
+import {mount, unmountAll} from "../../../utils/testing/mount.js";
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon-button/components/icon-button.test.js]: Happy dance

File: packages/wonder-blocks-icon-button/components/icon-button.test.js [hunk 7]
```diff
@@ -30,4 +37,61 @@ describe("IconButton", () => {
             ),
         ).toThrowError("Light is only supported for primary IconButtons");
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon-button/components/icon-button.test.js]: I wish this were using Arrange, Act, Assert. Having it all in an expect might be terse, but it's horrid.

File: packages/wonder-blocks-icon-button/components/icon-button.test.js [hunk 8]
```diff
@@ -30,4 +37,61 @@ describe("IconButton", () => {
...
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper
+            .find(`[data-test-id="icon-button"]`)
+            .first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(true);
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon-button/components/icon-button.test.js]: I'd love to see `// Arrange`, `// Act`, `// Assert` structure in these tests to make it clear what the test is doing.

File: packages/wonder-blocks-icon-button/components/icon-button.test.js [hunk 9]
```diff
@@ -30,4 +37,61 @@ describe("IconButton", () => {
...
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper
+            .find(`[data-test-id="icon-button"]`)
+            .first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(false);
```
Comment (somewhatabstract) [inline packages/wonder-blocks-icon-button/components/icon-button.test.js]: I'd love to see `// Arrange`, `// Act`, `// Assert` structure in these tests to make it clear what the test is doing.

File: packages/wonder-blocks-link/components/link.test.js [hunk 10]
```diff
@@ -1,12 +1,58 @@
...
+                </div>
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(true);
+    });
```
Comment (somewhatabstract) [inline packages/wonder-blocks-link/components/link.test.js]: I'd love to see `// Arrange`, `// Act`, `// Assert` structure in these tests to make it clear what the test is doing.

File: packages/wonder-blocks-link/components/link.test.js [hunk 11]
```diff
@@ -1,12 +1,58 @@
...
+                </div>
+            </MemoryRouter>,
+        );
+
+        expect(wrapper.find("#foo").exists()).toBe(false);
+        const buttonWrapper = wrapper.find(`[data-test-id="link"]`).first();
+        expect(buttonWrapper.exists()).toBe(true);
+        buttonWrapper.simulate("click", {button: 0});
+        expect(wrapper.find("#foo").exists()).toBe(false);
+    });
```
Comment (somewhatabstract) [inline packages/wonder-blocks-link/components/link.test.js]: I'd love to see `// Arrange`, `// Act`, `// Assert` structure in these tests to make it clear what the test is doing.

**Comment by kevinbarabash** [comment]
I can switch to using the triple-A structure.

**Pull request merged**